For some reason, the `test_hwo_isara_maint.py` gives the error below and core dumps with exit code -139:

```
2025-05-12 11:38:44,350 |WARNING| error polling Device(test/nodb/_isara) Powered attribute, retrying.
Traceback (most recent call last):
  File "/home/oscarsso/projects/mxcube/mxcubecore/mxcubecore/Command/Tango.py", line 274, in poll
    return read_attr()
           ^^^^^^^^^^^
  File "/home/oscarsso/projects/mxcube/mxcubecore/mxcubecore/Command/Tango.py", line 268, in read_attr
    value = self.raw_device.read_attribute(self.attribute_name).value
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oscarsso/micromamba/envs/mxcubeweb/lib/python3.11/site-packages/tango/green.py", line 208, in greener
    return executor.run(fn, args, kwargs, wait=wait, timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oscarsso/micromamba/envs/mxcubeweb/lib/python3.11/site-packages/tango/green.py", line 116, in run
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/oscarsso/micromamba/envs/mxcubeweb/lib/python3.11/site-packages/tango/device_proxy.py", line 550, in __DeviceProxy__read_attribute
    return __check_read_attribute(self._read_attribute(value, extract_as))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
PyTango.DevFailed: DevFailed[
DevError[
    desc = Failed to connect to device test/nodb/_isara
           The connection request was delayed.
           The last connection request was done less than 1000 ms ago
  origin = virtual void Tango::Connection::reconnect(bool) at (/src/cppTango/cppapi/client/devapi_base.cpp:712)
  reason = API_CantConnectToDevice
severity = ERR]

DevError[
    desc = Failed to read_attribute on device test/nodb/_isara, attribute Powered
  origin = virtual Tango::DeviceAttribute Tango::DeviceProxy::read_attribute(const string&) at (/src/cppTango/cppapi/client/devapi_base.cpp:5589)
  reason = API_AttributeFailed
severity = ERR]
]
Segmentation fault (core dumped)
```

@elmjag, @fabcor-maxiv do you see any reason for why this would happen ?

--- a/mxcubecore/BaseHardwareObjects.py
+++ b/mxcubecore/BaseHardwareObjects.py
@@ -61,7 +61,6 @@ if TYPE_CHECKING:
 
     from pydantic.v1 import BaseModel
 
-    from .CommandContainer import CommandObject
 
 __copyright__ = """ Copyright © 2010-2020 by the MXCuBE collaboration """
 __license__ = "LGPLv3+"
@@ -149,9 +148,16 @@ class ConfiguredObject:
     def get_by_id(self, _id: str) -> "ConfiguredObject":
         result = self
         for name in _id.split("."):
-            result = getattr(result, name)
+            # Roles are no longer added to HardwareObjects as attributes so
+            # we have to retreive them by role
+            result = result.get_object_by_role(name)
+
+            if not result:
+                result = getattr(result, name)
+
             if result is None:
                 break
+
         return result
 
     @property
@@ -331,19 +337,6 @@ class HardwareObjectNode:
     def __len__(self) -> int:
         warnings.warn("%s.__len__ is Deprecated. Avoid" % self.__class__.__name__)
         return sum(map(len, self.__objects))
-
-    def __setattr__(self, attr: str, value: Any) -> None:
-        try:
-            config = self.__dict__.get("_config")
-            if attr not in self.__dict__ and config and attr in config.model_dump():
-                warnings.warn(
-                    "%s.__setattr__ is Deprecated. Avoid" % self.__class__.__name__
-                )
-                self._set_property(attr, value)
-            else:
-                self.__dict__[attr] = value
-        except AttributeError:
-            self.__dict__[attr] = value
 
     def __getitem__(
         self,
@@ -1072,18 +1065,6 @@ class HardwareObject(ConfiguredObject, HardwareObjectNode, HardwareObjectMixin):
 
         obj = get_hardware_repository().get_hardware_object(name)
         self.__dict__.update(obj.__dict__)
-
-    def __getattr__(self, attr: str) -> Union["CommandObject", Any]:
-        if attr.startswith("__"):
-            raise AttributeError(attr)
-
-        try:
-            return CommandContainer.__getattr__(self, attr)
-        except AttributeError:
-            try:
-                return super().__getattr__(attr)
-            except AttributeError:
-                raise AttributeError(attr)
 
     def commit_changes(self) -> None:
         """Commit last changes back to configuration."""

--- a/mxcubecore/HardwareObjects/abstract/AbstractSampleView.py
+++ b/mxcubecore/HardwareObjects/abstract/AbstractSampleView.py
@@ -42,12 +42,15 @@ class AbstractSampleView(HardwareObject):
 
     def __init__(self, name):
         super().__init__(name)
-        self.camera = None
         self._focus = None
         self._zoom = None
         self._frontlight = None
         self._backlight = None
         self._shapes = None
+
+    @property
+    def camera(self):
+        return self.get_object_by_role("camera")
 
     @abc.abstractmethod
     def get_snapshot(

--- a/mxcubecore/HardwareObjects/abstract/sample_changer/Component.py
+++ b/mxcubecore/HardwareObjects/abstract/sample_changer/Component.py
@@ -7,7 +7,7 @@ class Component(object):
         self.container = container
         self.address = address
         self.scannable = scannable
-        self.id = None
+        self._id = None
         self.present = False
         self.selected = False
         self.scanned = False
@@ -26,7 +26,7 @@ class Component(object):
         Can be None if sample is unknown or not present
         :rtype: str
         """
-        return self.id
+        return self._id
 
     def get_address(self):
         """
@@ -123,8 +123,8 @@ class Component(object):
         Clears all sample info (also in components if object is a container)
         """
         changed = False
-        if self.id is not None:
-            self.id = None
+        if self._id is not None:
+            self._id = None
             changed = True
         if self.present:
             self.present = False
@@ -138,10 +138,10 @@ class Component(object):
     #########################           PROTECTED           #########################
     def _set_info(self, present=False, id=None, scanned=False):
         changed = False
-        if self.id != id:
-            self.id = id
+        if self._id != id:
+            self._id = id
             changed = True
-        if self.id:
+        if self._id:
             present = True
         if self.present != present:
             self.present = present

--- a/mxcubecore/HardwareObjects/mockup/ISPyBClientMockup.py
+++ b/mxcubecore/HardwareObjects/mockup/ISPyBClientMockup.py
@@ -29,6 +29,8 @@ class ISPyBClientMockup(ProposalTypeISPyBLims):
     def __init__(self, name):
         super().__init__(name)
 
+        self.loginType = LOGIN_TYPE_FALLBACK
+
     def init(self):
         try:
             self.base_result_url = self.get_property("base_result_url").strip()
@@ -67,9 +69,7 @@ class ISPyBClientMockup(ProposalTypeISPyBLims):
             "Laboratory": {"laboratoryId": 1, "name": "TEST eh1"},
         }
 
-    @property
-    def loginType(self):
-        return self.get_property("loginType", LOGIN_TYPE_FALLBACK)
+        self.loginType = self.get_property("loginType", LOGIN_TYPE_FALLBACK)
 
     def get_login_type(self):
         warnings.warn(

--- a/mxcubecore/HardwareRepository.py
+++ b/mxcubecore/HardwareRepository.py
@@ -169,6 +169,10 @@ def load_from_yaml(
 
     if not msg0:
         try:
+            config = configuration.pop("configuration", {})
+            # Set configuration with non-object properties.
+            result._config = result.HOConfig(**config)
+
             # Initialise object
             result._init()
         except Exception:
@@ -181,9 +185,6 @@ def load_from_yaml(
     if not msg0:
         # Recursively load contained objects (of any type that the system can support)
         objects = configuration.pop("objects", {})
-        config = configuration.pop("configuration", {})
-        # Set configuration with non-object properties.
-        result._config = result.HOConfig(**config)
 
         setup_commands_channels(result, configuration)
 

--- a/mxcubecore/HardwareRepository.py
+++ b/mxcubecore/HardwareRepository.py
@@ -293,7 +293,6 @@ def _attach_xml_objects(yaml_export_directory: Optional[Path], container, hwobj,
     container._roles.append(role)
     container._hwobj_by_role[role] = hwobj
 
-    setattr(container, role, hwobj)
     objects_by_role = hwobj._objects_by_role
     for role2, hwobj2 in objects_by_role.items():
         _attach_xml_objects(yaml_export_directory, hwobj, hwobj2, role2)

